internal/app/buildclient/artifact.go

```diff
@@ -17,6 +19,39 @@ import (
 	build "github.com/sylabs/scs-build-client/client"
 )
 
+// buildArtifact sends a build request for the specified arch, optionally publishing it to
+// libraryRef. Output is streamed to standard output. If the build cannot be submitted, or does not
+// succeed, an error is returned.
+func (app *App) buildArtifact(ctx context.Context, arch string, def []byte, buildContext string, libraryRef string) (*build.BuildInfo, error) {
+	opts := []build.BuildOption{build.OptBuildArchitecture(arch), build.OptBuildContext(buildContext)}
+	if libraryRef != "" {
+		opts = append(opts, build.OptBuildLibraryRef(libraryRef))
+	}
+
+	bi, err := app.buildClient.Submit(ctx, bytes.NewReader(def), opts...)
+	if err != nil {
+		return nil, fmt.Errorf("error submitting remote build: %w", err)
+	}
+	if err := app.buildClient.GetOutput(ctx, bi.ID(), os.Stdout); err != nil {
+		return nil, fmt.Errorf("error streaming remote build output: %w", err)
+	}
+	if bi, err = app.buildClient.GetStatus(ctx, bi.ID()); err != nil {
+		return nil, fmt.Errorf("error getting remote build status: %w", err)
+	}
+
+	// The returned info doesn't indicate an exit code, but a zero-sized image tells us something
+	// went wrong.
+	if bi.ImageSize() <= 0 {
+		return nil, errors.New("failed to build image")
+	}
+
+	if buildContext != "" {
+		_ = app.buildClient.DeleteBuildContext(ctx, buildContext)
```

Comment:
`buildArtifact` may be called multiple times with the same `buildContext` when multiple architectures are being built (https://github.com/tfrisch06/scs-build-client/blob/b42693c4a3be3c167a26dd9788ddab1c2652d18f/internal/app/buildclient/client.go#L236). In this case, the build context may be deleted before all architectures have finished, causing builds to fail.

This _probably_ belongs as a defer after `app.UploadBuildContext` succeeds (https://github.com/tfrisch06/scs-build-client/blob/b42693c4a3be3c167a26dd9788ddab1c2652d18f/internal/app/buildclient/client.go#L209)?